### PR TITLE
Added a Link struct to be used within the Movie struct

### LINKED ISSUE #69: Rework of the Movie.Links slice
As discussed in Discord the Links will now be saved as structs of the structure 

```
type link struct {
Name string
Link string
isSource bool
}
```

This change will require some work on the addMovie Handler.

This change is wanted to support new UI design decisions.

### PR DESCRIPTION
**Pull request by CptPie**
This PR closes #69 . (Sorry for the massive PR again)

I added a new file common/link.go for the new Link struct. This struct resembles the idea discussed in #69 :
```
type Link struct {
    Id       int
    Type     string
    Url      string
    IsSource bool
}
```
This struct also got a new Function called `l.DetermineType() string {...}` which looks the URL of the link struct and tries to determine the origin (IMDb or MyAnimeList) and returns the determined type.

I also added 4 new functions to the connector interface, namely AddTag, FindTag, GetTag, DeleteTag for the usual CRUD functionality. Of course i also implemented these Functions in the JSON Connector ready to use.
I also adapted the Movie struct to work with the Link struct as well as all the logic in `server.go`.

For Backwards compatibility of the database i also provide a migration Script `cmd/migrateLinks.go` which parses the data json file from the previous versions and converts the old Movie objects to new ones which use the Link struct described above.
For my own sanity i used an external library to not go insane trying to parse the old json format (i really hate the way go handles json in the standard library).
In a nutshell the script parses the old json data - deletes the movie map after saving it into ram and creates a temporary file. Afterwards the dataconnector is used to open this temporary file and add the new movies based on the old movies while also populating the new Link map.

Lastly i adapted the movie-info.html template to make use of the new fields introduced with the Link struct.
image

### CODE REVIEW

File: common/link.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
+package common
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
```
Comment (zorchenhimer) [inline common/link.go:8]: Use a `NewLink(url string) (*Link, error)` function here instead of requiring other bits of code to manually build the struct and call `ValidateLink()` and `DetermineLinkType()` directly.

**Review by zorchenhimer — CHANGES_REQUESTED** [review]

File: cmd/migrateLinks.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+		om := &OldMovie{}
+		mapstructure.Decode(movie, &om)
+		oldmovies = append(oldmovies, *om)
+	}
+
+	delete(fullData, "Movies")
+
+	data, err = json.MarshalIndent(fullData, "", " ")
+
+	err = ioutil.WriteFile("db/temp.json", data, 666)
```
[the remark this answers is not in the document]
Comment (CptPie) [reply cmd/migrateLinks.go:64]: Worked fine for me lul - yeah i'll change that.

File: cmd/migrateLinks.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+		// convert strings to link structs and add them to the db
+		links := []*mpc.Link{}
+		for id, linkUrl := range oldmovie.Links {
+			link := mpc.Link{
+				Url:      linkUrl,
+				IsSource: id == 0,
+			}
+
+			err := link.ValidateLink()
```
Comment (zorchenhimer) [inline cmd/migrateLinks.go:130]: these methods are no longer public.
